### PR TITLE
Bug fix/fix review comments

### PR DESCRIPTION
**Pull request by dothebart**
Address review comments of https://github.com/arangodb/arangodb/pull/19870
ep part: https://github.com/arangodb/enterprise/pull/1398

### CODE REVIEW

File: js/client/modules/@arangodb/aql-helper.js [hunk 1]
```diff
@@ -458,21 +447,67 @@ function unpackRawExpression (node, transform = false) {
...
+    'filtered',
+    'executionTime',
+    'httpRequests',
+    'fullCount',
+    'peakMemoryUsage',
+    'intermediateCommits'].forEach(item => { delete stats[item]; });
   return stats;
 }
 
+function execute_query(query, bindVars = null, options = {}) {
```
Comment (jsteemann) [inline js/client/modules/@arangodb/aql-helper.js:466]: I am generally unhappy that we now introduce yet another naming convention to our JavaScript code.
Our existing naming convention for JavaScript code is clearly `executeQuery` and not `execute_query`.

File: js/client/modules/@arangodb/aql-helper.js [hunk 2]
```diff
@@ -458,21 +447,67 @@ function unpackRawExpression (node, transform = false) {
...
+  let stmt = db._createStatement({query, bindVars: bindVars, count: true});
+  return stmt.execute();
+};
+
+function execute_all_json(plans, result, query) {
+  plans.forEach(function (plan) {
+    var jsonResult = db._executeJson(plan, {optimizer: {rules: ['-all']}});
```
Comment (jsteemann) [inline js/client/modules/@arangodb/aql-helper.js:473]: There is no need to ever use `var` in new JavaScript code.
```suggestion
    let jsonResult = db._executeJson(plan, {optimizer: {rules: ['-all']}});
```

File: js/client/modules/@arangodb/aql-helper.js [hunk 3]
```diff
@@ -458,21 +447,67 @@ function unpackRawExpression (node, transform = false) {
...
+    `;
+  }
+
+  let res = arango.POST("/_admin/execute", command);
+  return res;
+};
+
+function query_cache_properties_invalidate(cacheProperties = {}) {
+  let cachePropertiesStr = "";
+  if (cacheProperties !== {}) {
```
Comment (jsteemann) [inline js/client/modules/@arangodb/aql-helper.js:496]: This does not work. In JavaScript, `{}` is != `{}`:
```
$ node
Welcome to Node.js v16.13.1.
Type ".help" for more information.
> {} === {}
false
> {} !== {}
true
```

File: js/client/modules/@arangodb/aql-helper.js [hunk 4]
```diff
@@ -458,21 +447,67 @@ function unpackRawExpression (node, transform = false) {
...
+      return AQL_QUERY_CACHE_PROPERTIES(${JSON.stringify(arg)})
+    `;
+  } else {
+    command = `
+      return AQL_QUERY_CACHE_PROPERTIES()
+    `;
+  }
+
+  let res = arango.POST("/_admin/execute", command);
+  return res;
```
Comment (jsteemann) [inline js/client/modules/@arangodb/aql-helper.js:491]: This still relies on JavaScript running inside the server.
It should instead call the REST API `GET /_api/query/properties`.

File: js/client/modules/@arangodb/aql-helper.js [hunk 5]
```diff
@@ -458,21 +447,67 @@ function unpackRawExpression (node, transform = false) {
...
+  let cachePropertiesStr = "";
+  if (cacheProperties !== {}) {
+    cachePropertiesStr = JSON.stringify(cacheProperties);
+  }
+  let command = `
+    let cacheProperties = AQL_QUERY_CACHE_PROPERTIES(${cachePropertiesStr});
+    AQL_QUERY_CACHE_INVALIDATE();
+    return cacheProperties;
+  `;
+  return arango.POST("/_admin/execute", command);
```
Comment (jsteemann) [inline js/client/modules/@arangodb/aql-helper.js:504]: This still relies on JavaScript running inside the server.
It should instead use the REST API `PUT `/_api/query/properties`.

File: js/client/modules/@arangodb/arango-database.js [hunk 6]
```diff
@@ -986,6 +986,20 @@ ArangoDatabase.prototype._createStatement = function (data) {
...
+// //////////////////////////////////////////////////////////////////////////////
+
+ArangoDatabase.prototype._executeJson = function (plan, options = {}) {
+  let command = `
+        let data = ${JSON.stringify(plan)};
+        let opts = ${JSON.stringify(options)}
+        return AQL_EXECUTEJSON(data);
+      `;
+  return this._connection.POST("/_admin/execute", command).json;
+};
```
Comment (jsteemann) [inline js/client/modules/@arangodb/arango-database.js:1001]: This function should not be offered as part of our public API in arangosh. It is a function that is only used from tests and was never advertised.

Comment (dothebart) [reply js/client/modules/@arangodb/arango-database.js:1001]: we will eventually have to offer this API - at least for test purposes.

Comment (jsteemann) [reply js/client/modules/@arangodb/arango-database.js:1001]: We can move it into a helper file. But it should not appear as a suggestion when using auto-completion on the `db` object.

Comment (jsteemann) [reply js/client/modules/@arangodb/arango-database.js:1001]: Why not put this into `aql-helper.js` as some of the other test-only helper functions?

File: js/client/modules/@arangodb/aql-helper.js (hunk 3 above)
Comment (dothebart) [reply js/client/modules/@arangodb/aql-helper.js]: code obsoleted by direct API usage.

File: js/client/modules/@arangodb/aql-helper.js (hunk 1 above)
Comment (dothebart) [reply js/client/modules/@arangodb/aql-helper.js]: naming fixed.

**Pull request closed without merging**